### PR TITLE
Cherry pick and some manual fixes to address testing issues in 7.1

### PR DESCRIPTION
**Pull request by dragon512**
Just a note this should get all the tests passing again. However we still have fixes in master that are not in this, so expect some random time related failures in 7.1.x branch that will require to re-test AuTest to get everything to build.

I tried to get most of the fixes as Cherry picks, however in the end some fixes/cherry picks would pull in stuff extra stuff not in 7.1, For these I updated the extension manually form master to avoid pulling in extra stuff

### CODE REVIEW

**Comment by dragon512** [comment]
@jrushford This should fix the testing issues you had with your PR

**Comment by jrushford** [comment]
Thanks dragon512!

File: .gitignore [hunk 1]
```diff
@@ -164,3 +164,7 @@ tools/trafficserver.pc
 BUILDS
 DEBUG
 RELEASE
+
+# autest
+tests/env-test/
+tests/_sandbox/
```
Comment (SolidWallOfCode) [inline .gitignore]: Format: terminate last line in the file.

Comment (dragon512) [reply .gitignore]: fixed

**Comment by dragon512** [comment]
@zwoop This should be ready to go. Let me know if you have any concerns

**Pull request merged**